File: lib/socket_mobile_device.dart
import 'package:equatable/equatable.dart';

class SocketMobileDevice extends Equatable {
  /// friendly name
  final String name;

  /// unique ID, corresponding to the bluetooth address
  final String uuid;

  /// session ID
  final String guid;

  const SocketMobileDevice(this.name, this.uuid, this.guid);

  factory SocketMobileDevice.fromJson(Map json) {
    return SocketMobileDevice(
      json['name'],
      json['uuid'],
      json['guid'],
    );
  }

  @override
  List<Object> get props => [uuid];
}

File: android/src/main/java/co/eivo/socket_mobile/SocketMobileDevice.java
package co.eivo.socket_mobile;

import java.util.HashMap;

public class SocketMobileDevice {

    public String name;
    public String uuid;
    public String guid;

    SocketMobileDevice(String name, String uuid, String guid) {
        this.name = name;
        this.uuid = uuid;
        this.guid = guid;
    }

    public HashMap<String, String> toMap() {
        final HashMap<String, String> map = new HashMap<String, String>();
        map.put("name", this.name);
        map.put("uuid", this.uuid);
        map.put("guid", this.guid);

        return map;
    }
}